sia-gpu-miner.c

```diff
@@ -71,47 +77,43 @@ double grindNonces(size_t items_per_iter, int cycles_per_iter) {
 		}
 		target_corrupt_flag = 1;
 		printf("Received corrupt target from Sia\n");
-		printf("Usually this resolves itself within a minute or so\n");
-		printf("If it happens frequently trying increasing seconds per iteration\n");
-		printf("e.g. \"./gpu-miner -s 3 -c 200\"\n");
 		printf("Waiting for problem to be resolved...");
 		fflush(stdout);
+		return -1;
 	}
 	target_corrupt_flag = 0;
+	size_t global_item_size = 1 << intensity;
+
+	// Copy target to header
+	for (i = 0; i < 8; i++) {
+		blockHeader[i + 32] = target[7-i];
+	}
```

Comment:
it's an optimization. Only the first 6 bytes of the target matter, and the other 2 are a buffer. We won't need more than that unless the hashrate increases by about 100,000x the current hashrate, at which point we'll just switch from looking at the first 8 bytes to looking at bytes 4-12.